Tests/Pcap++Test/Tests/FileTests.cpp

```diff
@@ -99,21 +100,78 @@ PTF_TEST_CASE(TestPcapFileReadWrite)
+	for (auto const path : { EXAMPLE_PCAP_MICRO_PATH, EXAMPLE_PCAP_MICRO_WRITE_PATH })
+	{
+		// Read micro precision file
+		pcpp::PcapFileReaderDevice readerDevMicro(path);
+		PTF_ASSERT_EQUAL(readerDevMicro.getTimestampPrecision(), pcpp::FileTimestampPrecision::Unknown, enumclass);
+		PTF_ASSERT_TRUE(readerDevMicro.open());
+		PTF_ASSERT_EQUAL(readerDevMicro.getTimestampPrecision(),
+		                 pcpp::PcapFileReaderDevice::isNanoSecondPrecisionSupported()
+		                     ? pcpp::FileTimestampPrecision::Nanoseconds
+		                     : pcpp::FileTimestampPrecision::Microseconds,
+		                 enumclass);
+
+		pcpp::RawPacket readPacketNano2, readPacketMicro2;
+		PTF_ASSERT_TRUE(readerDevMicro.getNextPacket(readPacketMicro2));
+		PTF_ASSERT_EQUAL(readPacketMicro2.getPacketTimeStamp().tv_sec, 1);
+		PTF_ASSERT_EQUAL(readPacketMicro2.getPacketTimeStamp().tv_nsec, 2000);
+
+		PTF_ASSERT_TRUE(readerDevMicro.getNextPacket(readPacketNano2));
+		PTF_ASSERT_EQUAL(readPacketNano2.getPacketTimeStamp().tv_sec, 1);
+		PTF_ASSERT_EQUAL(readPacketNano2.getPacketTimeStamp().tv_nsec, 1000);
+		readerDevMicro.close();
+	}
+}  // TestPcapFileMicroPrecision
 
+PTF_TEST_CASE(TestPcapFileNanoPrecision)
+{
 	// Writer precision support should equal to reader precision support
 	PTF_ASSERT_EQUAL(pcpp::PcapFileWriterDevice::isNanoSecondPrecisionSupported(),
 	                 pcpp::PcapFileReaderDevice::isNanoSecondPrecisionSupported());
 
+	if (!pcpp::PcapFileWriterDevice::isNanoSecondPrecisionSupported())
+	{
+		std::cout << "Pcap nano precision is not supported on the current platform! "
+		             "The test will check for proper failure messages.\n";
+		PTF_ASSERT_RAISES(pcpp::PcapFileWriterDevice(EXAMPLE_PCAP_NANO_WRITE_PATH, pcpp::LINKTYPE_ETHERNET, true),
+		                  std::runtime_error,
+		                  "PcapPlusPlus was compiled without nano precision support which requires libpcap > 1.5.1. "
+		                  "Please recompile PcapPlusPlus with nano precision support to use this feature.");
+
+		pcpp::PcapFileReaderDevice readerDevNano(EXAMPLE_PCAP_NANO_PATH);
+		PTF_ASSERT_FALSE(readerDevNano.open());
+		return;
+	}
+
+	std::array<uint8_t, 16> testPayload = { 0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07,
+		                                    0x08, 0x09, 0x0A, 0x0B, 0x0C, 0x0D, 0x0E, 0x0F };
+
+	pcpp::RawPacket rawPacketMicro(testPayload.data(), testPayload.size(), timeval({ 1, 2 }), false);     // 1.000002000
+	pcpp::RawPacket rawPacketNano(testPayload.data(), testPayload.size(), timespec({ 1, 1234 }), false);  // 1.000001234
+
 	// Write nano precision file
-	pcpp::Logger::getInstance().suppressLogs();
-	pcpp::PcapFileWriterDevice writerDevNano(EXAMPLE_PCAP_NANO_PATH, pcpp::LINKTYPE_ETHERNET, true);
-	pcpp::Logger::getInstance().enableLogs();
+	pcpp::PcapFileWriterDevice writerDevNano(EXAMPLE_PCAP_NANO_WRITE_PATH, pcpp::LINKTYPE_ETHERNET, true);
 	PTF_ASSERT_EQUAL(writerDevNano.getTimestampPrecision(),
```

Comment:
If we got here, it means that `pcpp::PcapFileWriterDevice::isNanoSecondPrecisionSupported()` is `true` so the precision should always be `pcpp::FileTimestampPrecision::Nanoseconds`. Same for line 181-185